File: app/src/main/java/com/hulkdx/moneymanagerv2/injection/module/ActivityModule.java
/**
 * Created by Mohammad Jafarzadeh Rezvan on 7/6/2017.
 */

package com.hulkdx.moneymanagerv2.injection.module;

import android.app.Activity;
import android.content.Context;

import dagger.Module;
import dagger.Provides;
import com.hulkdx.moneymanagerv2.injection.ActivityContext;

@Module
public class ActivityModule {

    private Activity mActivity;

    public ActivityModule(Activity activity) {
        mActivity = activity;
    }

    @Provides
    Activity provideActivity() {
        return mActivity;
    }

    @Provides
    @ActivityContext
    Context providesContext() {
        return mActivity;
    }
}

File: presentation/src/main/java/com/hulkdx/moneymanagerv2/di/modules/ActivityModule.kt
package com.hulkdx.moneymanagerv2.di.modules

import android.app.Activity
import android.content.Context

import dagger.Module
import dagger.Provides
import hulkdx.com.domain.di.ActivityContext

/**
 * Created by Mohammad Jafarzadeh Rezvan on 09/11/2018.
 */

@Module
class ActivityModule(private val mActivity: Activity) {

    @Provides
    fun provideActivity(): Activity = mActivity

    @Provides
    @ActivityContext
    fun providesContext(): Context = mActivity
}
